contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -78,17 +79,34 @@ public String getTypeName() {
   }
 
   @Override
-  public List<Pair<String, ? extends Table>> getTablesByNamesByBulkLoad(final List<String> tableNames) {
+  public List<Pair<String, ? extends Table>> getTablesByNamesByBulkLoad(final List<String> tableNames, final int bulkSize) {
+    final int totalTables = tableNames.size();
     final String schemaName = getName();
-    final List<Pair<String, ? extends Table>> tableNameToTable = Lists.newArrayList();
-    List<org.apache.hadoop.hive.metastore.api.Table> tables;
-    try {
-      tables = DrillHiveMetaStoreClient.getTableObjectsByNameHelper(mClient, schemaName, tableNames);
-    } catch (TException e) {
-      logger.warn("Exception occurred while trying to list tables by names from {}: {}", schemaName, e.getCause());
-      return tableNameToTable;
+    final List<org.apache.hadoop.hive.metastore.api.Table> tables = Lists.newArrayList();
+
+    // In each round, Drill asks for a sub-list of all the requested tables
+    for(int fromIndex = 0; fromIndex < totalTables; fromIndex += bulkSize) {
+      final int toIndex = Math.min(fromIndex + bulkSize, totalTables);
+      final List<String> eachBulkofTableNames = tableNames.subList(fromIndex, toIndex);
+      List<org.apache.hadoop.hive.metastore.api.Table> eachBulkofTables;
+      // Retries once if the first call to fetch the metadata fails
+      synchronized(mClient) {
```

Comment:
This is refactored code from the fix for DRILL-4577. (https://github.com/apache/drill/pull/461)
I didn't really change it.
Going thru the code, it appears that m_client may be cached and reused and so probably should be synchronized.